server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/MinorCompactor.java

```diff
@@ -131,12 +134,21 @@ public CompactionStats call() {
           reportedProblem = true;
         } catch (RuntimeException | NoClassDefFoundError e) {
           // if this is coming from a user iterator, it is possible that the user could change the
-          // iterator config and that the
-          // minor compaction would succeed
+          // iterator config and that the minor compaction would succeed
+          // If the minor compaction stalls for too long during recovery, it can interfere with
+          // other tables loading
+          // Throw exception if this happens so assignments can be rescheduled.
+          if (retryCounter >= 4 && mincReason.equals(MinorCompactionReason.RECOVERY)) {
+            log.warn("Minc is stuck for too long during recovery, throwing error for reschedule.");
```

Comment:
Including the key extent in log messages can be invaluable for debugging.  The existing log messages around failures don't include the extent.

Reply:
Good idea, I will revise the logging.

Reply:
It does seem that the warn/error messaging in `AssignmentHandler` and `MinorCompationTask` does include the key extent in the logging. I can still include it in this log message to help track it. I do worry about log overload throughout the whole process of rescheduling but I am not sure if that is an actual issue or not.